lib/katello/engine.rb

```diff
@@ -250,6 +250,10 @@ def self.with_remote_execution?
     (RemoteExecutionFeature rescue false) ? true : false
   end
 
+  def self.with_katello_agent?
+    SETTINGS[:katello][:agent][:enabled]
```

Comment:
You may want to add a default here https://github.com/Katello/katello/pull/9178/files#diff-7dc78088a8bc944d190599ecf572ea01ab9b85fc91415ddcadf17dd5e9b1638eR59

You might consider using dig here as well to future proof it if this setting got removed or wasn't present:

```
SETTINGS.dig(:katello, :agent, :enabled)
```